TSG/Update/cau-debug-in-23h2.md

```diff
@@ -0,0 +1,263 @@
+
+## Log analysis
+
+To diagnose and investigate problems, there are two possible scenarios:
+
+### Kusto
+
+If logs were uploaded to kusto (using `Send-DiagnosticData` or proactive log collection), we can get details as seen below. You can use the following queries to narrow down the relevant logs:
+
+![Image showing example output from Send-DiagnosticData PowerShell command](./CauHelpSendDiagnosticData.png)
+
+- To list all file names which contain a particular match string
+
+    > let resUri = @"";
+    > TextLogs  
+    > | where AEODeviceARMResourceUri  == resUri
+    > | where C_FileLinesContent contains @"<mark><your_key_string_here></mark>"
+    > | where C_FileName contains @"<mark><see_prefixes_below></mark>"
+    > | distinct C_FileName
+
+    Here,
+
+    - C_FileLinesContent can be
+        - "<mark>Import-module clusterawareupdating; Invoke-CauRun</mark>" to find all sessions
+        - <mark>RunId</mark> to find all related files
+    - C_FileName can be one of the following prefixes (for the respective type of file)
+        - "<mark>debugTrace</mark>_" or "<mark>cauwmi</mark>" or "<mark>cauReport</mark>-" 
+        - Or the suffix “cluster.log” or “cluster.txt” for cluster logs
+    - Example (to find debugTrace files for a given runID)
+
+    > let resUri = @" /Subscriptions/6a260d98-e0de-4b8d-9f3a-b7c10c8704df/resourceGroups/EDGECI-REGISTRATION-HC1n42r1602-4RAvozGK/providers/Microsoft.AzureStackHCI/clusters/s-cluster";
+    > TextLogs  
+    > | where AEODeviceARMResourceUri  == resUri
+    > | where C_FileLinesContent contains @"43703982-acaf-496d-bf20-97f859219631"
+    > | where C_FileName contains @"debugTrace_"
+    > | distinct C_FileName
+
+- Using the file name from above (paste into dirpath), get the contents of file using the following query (note that this also prints a lot of noise):
+
+    > let dirpath = @""; // just add these two
+    > let resUri = @"";
+    > TextLogs  
+    > | where AEODeviceARMResourceUri  == resUri
+    > | where C_FileName startswith dirpath
+    > | extend C_SequenceNumber = toint(C_SequenceNumber)
+    > | order by C_SequenceNumber asc
+    > | summarize allcontent = strcat_array(make_list(C_FileLinesContent), "")
+ 
+- To remove all the noise and just print relevant content use the following query
+
+    > let dirpath = @""; // just add these two
+    > let resUri = @"";
+    > TextLogs  
+    > | where AEODeviceARMResourceUri  == resUri
+    > | where C_FileName startswith dirpath
+    > | extend C_SequenceNumber = toint(C_SequenceNumber)
+    > | order by C_SequenceNumber asc  
+    > | extend C_FileLinesContent = split(C_FileLinesContent, "\n")
+    > | mv-expand C_FileLinesContent to typeof(string)
+    > | extend C_FileLinesContent = extract("\"(.*)", 1, C_FileLinesContent)
+    > | summarize make_list(C_FileLinesContent)
+
+- See example below
+
+    ![Example Kusto query showing the content of the CAU debug trace](./CauHelpKusto.png)       
```

Comment:
Please remove references to internal Kusto queries since this TSG is meant for customer self-remediation.